feat: added API calls for the process annotation

vdiezel opened the pull request:
BREAKING CHANGE: Forcing major version bump because it failed previously; Dropped support for Node v14;

Commit:
feat: added API calls for the process annotation

FelixMZ2018 reviewed (changes requested):
Are we still updating the readme ? If so add it there as well. 

Also annoyingly the last breaking change did not trigger a release so this has to be a major release

Commit:
docs: adds documentation for the process annotation API

BREAKING CHANGE: triggering major version update to catch up with previously failed version bump

vdiezel committed:
Merge branch 'main' into add-annotation-endpoints

FelixMZ2018 reviewed (approved):
lgtm! Could you add the dropped node 14 support in the changelog as well please ?
